core/client/fs/src/main/java/alluxio/client/file/cache/CacheManager.java

```diff
@@ -18,7 +18,35 @@
 /**
  * Interface for managing cached pages.
  */
-public interface CacheManager extends AutoCloseable  {
+public interface CacheManager extends AutoCloseable {
+
+  /**
+   * State of a cache.
+   */
+  enum State {
+    /**
+     * this cache is read only.
+     */
+    READ_ONLY(0),
```

Comment:
If `restoreOrInit` hasn't completed, is it safe for `get` calls? Will there be some objects which could be null?

Reply:
protected by locks